### ISSUE TITLE
Size of QR code

### ISSUE THREAD

**Issue by detrin** [dtonon/papiro#2]
Hi, I wonder whether it would be possible to change the amount of information put into QR codes. Let's say that the printed paper degrades over time and printing small squares in QR code will make it unreadable in some time. Would you consider adding another parameter into shell script that will set the maximum size of QR code with selected correction level?

**Comment by dtonon** [comment]
This is an interesting matter.

The error correction is already possibile with the `-l` flag and the standard value is Low. Increasing it to H (Hight) should give a quite good resistance against degradation. Did you make a test with it, try to ruin the paper and test a recovery?

Currently Papiro uses the wider qrcode version (v40) in a 3x4 grid, because this fits well in the space and seems to strike a good balance between data density and the quantity of sheets needed. I could add a 2X3 or 1x2 options to let the user increase the single qrcodes size and so further improve the resistance.

**Comment by detrin** [comment]
> Did you make a test with it, try to ruin the paper and test a recovery?

No, but two years I was basically writing my own version of your papiro. I found out that the print quality differs. FInding the papers with QR codes now, I noticed that some of them started to degrade. So, this is not a theoretical issue.

> Currently Papiro uses the wider qrcode version (v40) in a 3x4 grid, because this fits well in the space and seems to strike a good balance between data density and the quantity of sheets needed. I could add a 2X3 or 1x2 options to let the user increase the single qrcodes size and so further improve the resistance.

Okay, would you consider adding apart from quality of error correction also maximal size of QR codes?

**Comment by dtonon** [comment]
I will check this in the following days.
